### DISCUSSION TITLE
Does the moquette supports MQTT 5? (question)

### DISCUSSION THREAD

**Discussion by BigGitWorld** [moquette-io/moquette#784] [Q&A]
Hello.
Does the latest version of Moquette (0.17) support MQTT version 5?

**Comment by oliverunger** [comment]
Looking into the code at class MQTTConnection it does not.
```
if (isNotProtocolVersion(msg, MqttVersion.MQTT_3_1) && isNotProtocolVersion(msg, MqttVersion.MQTT_3_1_1)) {
            LOG.warn("MQTT protocol version is not valid. CId: {}", clientId);
            abortConnection(CONNECTION_REFUSED_UNACCEPTABLE_PROTOCOL_VERSION);
            return PostOffice.RouteResult.failed(clientId);
        }
```
I have to find something else too. But testcontainers are too heavy.

**Comment by andsel** [comment]
Not yet, I'm working on pushing it in spare time, the version 0.18 is the target for MQTT5 support.
Keep an eye to #695 to check where we are.
